clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -55,6 +56,18 @@ export class HttpConnection implements IConnection {
         options = options || {};
         options.accessTokenFactory = options.accessTokenFactory || (() => null);
         options.logMessageContent = options.logMessageContent || false;
+        options.WebSocket = options.WebSocket;
+        options.EventSource = options.EventSource;
+
+        if (typeof window !== "undefined") {
```

Comment:
I have yet a third way to do this that might work better and won't need this funky `window` or `global` thing ;)

```javascript
if(typeof WebSocket !== "undefined") {
    options.WebSocket = options.WebSocket || WebSocket
}
if(typeof EventSource !== "undefined") {
    options.EventSource = options.EventSource || EventSource 
}
```

Then your tests can go back (yet again) to `global["WebSocket"] = blah`